hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -834,11 +837,11 @@ public void filterRelevantSstFiles(Set<String> inputFiles,
               .constructBucketKey(new String(iterator.key(), UTF_8));
           if (!RocksDiffUtils
               .isKeyWithPrefixPresent(prefix, firstKey, lastKey)) {
-            inputFiles.remove(filename);
+            fileIterator.remove();
```

Comment:
Do we have some UT to cover this ?

Reply:
IIRC @sadanand48 had a UT that found this bug in the first place.

Reply:
Added a unit test.